contrib/llvm-cmake/CMakeLists.txt

```diff
@@ -2,11 +2,12 @@
 # tools to be build for host architecture and everything else for target architecture (e.g. AArch64)
 # Possible workaround is to use llvm-tblgen from some package...
 # But lets just enable LLVM for native builds
-if (CMAKE_CROSSCOMPILING OR SANITIZE STREQUAL "undefined")
+if (NOT (${CMAKE_SYSTEM_NAME} MATCHES ${CMAKE_HOST_SYSTEM_NAME}) OR SANITIZE STREQUAL "undefined")
```

Comment:
CMAKE_CROSSCOMPILING can be true even without cross compilation
```
This variable will be set to true by CMake if the [CMAKE_SYSTEM_NAME]
(https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html#variable:CMAKE_SYSTEM_NAME) 
variable has been set manually (i.e. in a toolchain file or as a cache entry from the [cmake]
```
https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html